vm/systemSmartContracts/staking.go

```diff
@@ -567,10 +677,15 @@ func (r *stakingSC) slash(args *vmcommon.ContractCallInput) vmcommon.ReturnCode
 		return vmcommon.UserError
 	}
 
+	if registrationData.UnJailedNonce >= registrationData.JailedNonce {
+		r.addToJailedNodes()
```

Comment:
not the other way around? registrationData.Jailed > Unjailed?

Reply:
if you already unjailed yourself a few moments ago there is no need to remove that again.

Reply:
do you have to compute everything else if you did it a few moments ago?